tests/nnstreamer_decoder_tensor_region/unittest_tensor_region.cc

```diff
@@ -0,0 +1,209 @@
+ *
+ * @param bus The GStreamer bus.
+ * @param message The GStreamer message.
+ * @param data Pointer to the TensorRegion structure.
+ * @return gboolean Returns TRUE to continue receiving messages.
+ */
+static gboolean
+on_pipeline_message (GstBus *bus, GstMessage *message, TensorRegion *data)
+{
+  switch (GST_MESSAGE_TYPE (message)) {
+    case GST_MESSAGE_EOS:
+      break;
+    case GST_MESSAGE_ERROR:
+      {
+        g_print ("Received error\n");
+
+        GError *err = NULL;
+        gchar *dbg_info = NULL;
+
+        gst_message_parse_error (message, &err, &dbg_info);
+        g_printerr ("ERROR from element %s: %s\n",
+            GST_OBJECT_NAME (message->src), err->message);
+        g_printerr ("Debugging info: %s\n", (dbg_info) ? dbg_info : "none");
+        g_error_free (err);
+        g_free (dbg_info);
+      }
+
+      break;
+    case GST_MESSAGE_STATE_CHANGED:
+      break;
+    default:
+      break;
+  }
+
+  /** Return FALSE to stop receiving messages after the callback function
+   * has handled the current message. */
+  return G_SOURCE_CONTINUE;
+}
+
+
+/**
+ * @brief Test for tensor_decoder::tensor_region
+ */
+TEST (tensorDecoder, tensorRegion)
+{
+  GstBus *bus;
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  if (root_path == nullptr)
+    root_path = "..";
+
+  const gchar *tensor_0 = g_build_filename (root_path, "tests",
+      "nnstreamer_decoder_tensor_region", "mobilenet_ssd_tensor.0", nullptr);
+  const gchar *tensor_1 = g_build_filename (root_path, "tests",
+      "nnstreamer_decoder_tensor_region", "mobilenet_ssd_tensor.1", nullptr);
+  const gchar *labels_path = g_build_filename (
+      root_path, "tests", "test_models", "labels", "labels.txt", nullptr);
+  const gchar *box_priors_path = g_build_filename (root_path, "tests",
+      "nnstreamer_decoder_boundingbox", "box_priors.txt", nullptr);
+
+  ASSERT_TRUE (g_file_test (tensor_0, G_FILE_TEST_EXISTS));
+  ASSERT_TRUE (g_file_test (tensor_1, G_FILE_TEST_EXISTS));
+  ASSERT_TRUE (g_file_test (labels_path, G_FILE_TEST_EXISTS));
+  ASSERT_TRUE (g_file_test (box_priors_path, G_FILE_TEST_EXISTS));
+
+  /** Create the GStreamer pipeline */
+  gchar *pipeline_str = g_strdup_printf ("multifilesrc name=fs1 location=%s start-index=0 stop-index=1 caps=application/octet-stream ! tensor_converter name=el1 input-dim=4:1:1917:1 input-type=float32 ! mux.sink_0 \
+       multifilesrc name=fs2 location=%s start-index=0 stop-index=1 caps=application/octet-stream ! tensor_converter name=el2 input-dim=91:1917:1 input-type=float32 ! mux.sink_1 \
+       tensor_mux name=mux ! other/tensors,format=static ! tensor_decoder mode=tensor_region option1=1 option2=%s option3=%s ! appsink name=sinkx ",
+      tensor_0, tensor_1, labels_path, box_priors_path);
+
+  GstElement *pipeline = gst_parse_launch (pipeline_str, nullptr);
+  g_free (pipeline_str);
+
+  GstElement *app_sink = gst_bin_get_by_name (GST_BIN (pipeline), "sinkx");

```

Comment:
Caller owns returned reference. Please unref the element after use.

Reply:
Thank you, I have unrefed the app_sink at the end.